frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,154 @@
+      organization: ['', Validators.required],
+      startedAt: ['', Validators.required],
+      actionOngoing: ['', Validators.required],
+      finishedAt: ['', Validators.required],
+      constantCoop: [],
+      recruitmentStartDate: ['', Validators.required],
+      recruitmentEndDate: ['', Validators.required],
+      volunteersLimit: [],
+      reserveRecruitmentStartDate: [],
+      reserveRecruitmentEndDate: [],
+      reserveVolunteersLimit: [],
+      description: ['', Validators.required],
+      timeCommitment: ['', Validators.required],
+      benefits: ['', Validators.required],
+      requirements: ['', Validators.required],
+      image: [],
+    })
+
+      this.userSubscription = this.authService.user$
+      .subscribe(
+        response => {
+          this.user = response;
+            this.isAdmin = this.user['isAdministrator'];
+            this.hasOrganization = this.user['organizations'].length > 0;
+          }
+      );
+
+    this.route.params
+    .map(params => params.offerId)
+    .filter(offerId => offerId !== undefined)
+    .switchMap(offerId => this.offersService.getOffer(offerId))
+    .do(offer => this.toDataUrl(offer.image, (base64) => this.offer.image = {
+      content: base64,
+      filename: 'image.jpg'
+    }))
+    .subscribe((response: BaseOffer) => {
+      this.offer = response as AppOffer;
+      this.inEditMode = true;
+      this.form.patchValue(this.offer);
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.userSubscription) {
+      this.userSubscription.unsubscribe();
+    }
+  }
+
+  toDataUrl(url, callback) {
+    const xhr = new XMLHttpRequest();
+    xhr.onload = function() {
+        const reader = new FileReader();
+        reader.onloadend = function() {
+            callback(reader.result);
+        }
+        reader.readAsDataURL(xhr.response);
+    };
+    xhr.open('GET', url);
+    xhr.responseType = 'blob';
+    xhr.send();
+}
+
+  onSubmit(offer: AppOffer) {
+    // TODO - delete those when we decide what date format we want to have
+    offer.startedAt = offer.startedAt + 'T00:00:00Z';
+    offer.finishedAt = offer.finishedAt + 'T00:00:00Z';
+
+    if (!offer.recruitmentEndDate) {
+      offer.recruitmentEndDate = null
+    } else {
+      offer.recruitmentEndDate = offer.recruitmentEndDate + 'T00:00:00Z';
+    }
+    if (!offer.recruitmentStartDate) {
+      offer.recruitmentStartDate = null
+    } else {
+      offer.recruitmentStartDate = offer.recruitmentStartDate + 'T00:00:00Z';
+    }
+    if (!offer.reserveRecruitmentEndDate) {
+      offer.reserveRecruitmentEndDate = null
+    } else {
+      offer.reserveRecruitmentEndDate = offer.reserveRecruitmentEndDate + 'T00:00:00Z';
+    }
+    if (!offer.reserveRecruitmentStartDate) {
+      offer.reserveRecruitmentStartDate = null
+    } else {
+      offer.reserveRecruitmentStartDate = offer.reserveRecruitmentStartDate + 'T00:00:00Z';
+    }
+
+    this.form.value.image = this.offer.image;
```

Comment:
We need to transform this.offer.image.content before assigning it to this.form.value.image.

Now it looks something like this: 
`"data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA
AAAFCAYAAACNbyblAAA=="` and this is for the browser to render it properly, however we need for backend to accept we need to send raw content data from this string. To to that we need to remove `"data:image/png;base64, `
and send only base64 encoded string `iVBORw0KGgoAAAANSUhEUgAAAAUA
AAAFCAYAAACNbyblAAA=="`.

Solution with `this.offer.image.content.replace( // replacement strings here )` should be enough, although I might be wrong. Try to find some examples on the Internet on how to use JavaScripts String.replace() method :)